cdci_data_analysis/configurer.py

```diff
@@ -33,89 +33,62 @@
 
 logger = logging.getLogger("conf")
 
-
-# TODO: this looks rather specific to INTEGRAL?
-
 class DataServerConf:
 
-    def __init__(self, data_server_url,
-                 data_server_remote_cache=None,
-                 dispatcher_mnt_point=None,
-                 dummy_cache=None,
-                 products_url=None,
-                 data_server_port=None):
-
-        if data_server_port is not None:
-            logger.warning(
-                "data_server_port is disregarded, since it is naturally included in the url")
-
-        logger.info("building config from %s %s %s", data_server_url,
-                    data_server_remote_cache, dispatcher_mnt_point)
+    def __init__(self, **kwargs):
 
-        if data_server_url is None:
-            logger.warning(
-                f"problem constructing {self}: data_server_url is None: {data_server_url}")
+        conf = kwargs.copy()
 
-        if data_server_remote_cache is None:
-            logger.warning(
-                f"problem constructing {self}: data_server_remote_cache is None: {data_server_remote_cache}")
+        logger.info("building config from %s", conf)
 
-        if dispatcher_mnt_point is None:
-            logger.warning(
-                f"problem constructing {self}: dispatcher_mnt_point is None: {dispatcher_mnt_point}")
+        required_keys = ['data_server_url']
 
-        self.data_server_url = data_server_url
+        for key in required_keys:
+            try:
+                self.__setattr__(key, conf.pop(key))
+            except KeyError as e:
+                logger.error(
+                    f"problem constructing {self}: {key} configuration key is required")
+                raise e
 
-        # dummy prods local cache
-        self.dummy_cache = dummy_cache
+        if conf.pop('data_server_port', None) is not None:
+             logger.warning(
+                 "data""_server_port is disregarded, since it is naturally included in the url")
 
-        # path to dataserver cache
-        self.data_server_remote_path = data_server_remote_cache
+        #special cases (maybe INTEGRAL specific a bit)
+        if conf.get('data_server_remote_cache', None) is not None:
+            # path to dataserver cache
+            self.data_server_remote_path = conf.pop('data_server_remote_cache')
+        else:
+            self.data_server_remote_path = None
 
-        if dispatcher_mnt_point is not None:
-            self.dispatcher_mnt_point = os.path.abspath(dispatcher_mnt_point)
+        if conf.get('dispatcher_mnt_point', None) is not None:
+            self.dispatcher_mnt_point = os.path.abspath(conf.pop('dispatcher_mnt_point'))
             FilePath(file_dir=self.dispatcher_mnt_point).mkdir()
         else:
             self.dispatcher_mnt_point = None
 
-        self.products_url = products_url
-
         if self.dispatcher_mnt_point is not None and self.data_server_remote_path is not None:
```

Comment:
Ironically considering the intent to reduce instrument dependence, the only parameter that are still treated explicitly are INTEGRAL (clearly since special treatment had to be preserved, unfortunatly, at least for now)